### PR TITLE
feat/Initialize Spring Boot project with JWT authentication

### PR DESCRIPTION
**Pull request by blackspyek**
# Pull Request: feat/Initialize Spring Boot project with JWT authentication

## Description

This pull request introduces the foundational setup for a Spring Boot project with JWT-based authentication. The following features have been implemented:

- **Spring Boot Project Initialization**: Set up the basic structure of the Spring Boot application.
- **JWT Authentication**: Implemented JWT-based authentication for user login and registration.
- **Authentication Service**: Created a `AuthenticationService` to manage user registration, authentication.
- **User Controller**: Added a `UserController` to handle user role change endpoints.
- **Auth Controller**: Added a `AuthController` to handle user register and login endpoints.
- **Role Management**: Implemented custom validation for role assignments.
- **Error Handling**: Introduced centralized error handling using `ControllerAdvice`.
- **Controller Advice**: Added global exception handling for cleaner and more maintainable error responses.

## How to Test

1. Clone the repository.
2. Run the Spring Boot application.
3. Create ,env
3. Run the following command to start the application using Docker Compose:
```bash
docker compose up
```
5. Use Postman or a similar tool to test the following API endpoints:
   - **POST /api/auth/register**: Register a new user.
   - **POST /api/auth/login**: Login with the user credentials to receive a JWT token.
   - **PUT /api/user/roles/{username}**: Login with the user credentials to receive a JWT token.
6. Ensure that role validation is applied correctly, and error handling works as expected for invalid inputs.

## Notes

- JWT tokens are used for authentication and must be included in the `Authorization` header as `Bearer <token>` for protected routes.
- User has autogenerated username
- Error handling is centralized, ensuring that consistent and clear error responses are returned for common issues.
- phpMyAdmin is available at port :5550

### CODE REVIEW

File: library/src/main/java/org/pollub/library/utils/ValidRoleSet.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import jakarta.validation.Constraint;
+import jakarta.validation.Payload;
+
+import java.lang.annotation.*;
+
+@Documented
+@Constraint(validatedBy = ValidRoleSetValidator.class)
+@Target({ElementType.FIELD, ElementType.PARAMETER})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface ValidRoleSet {
```
Comment (DFraczek) [inline library/src/main/java/org/pollub/library/utils/ValidRoleSet.java:12]: I don't know what's going on here

Comment (DFraczek) [reply library/src/main/java/org/pollub/library/utils/ValidRoleSet.java:12]: Whats the purpose of this

Comment (blackspyek) [reply library/src/main/java/org/pollub/library/utils/ValidRoleSet.java:12]: Jakarta Bean Validation (and similar validation frameworks) don't have built-in support for validating custom enums, like role sets. So, you created this custom annotation to apply your own validation logic (inside the ValidRoleSetValidator class) for roles, ensuring that only valid roles are allowed in the system.

In summary, this annotation allows you to easily mark fields or parameters in your application that need role validation, and it delegates the actual validation logic to the ValidRoleSetValidator class.

Comment (DFraczek) [reply library/src/main/java/org/pollub/library/utils/ValidRoleSet.java:12]: image

**Review by DFraczek — APPROVED** [review]
LGTM

**Pull request merged**